### PR TITLE
[PROCS-4000] Update CRD to add new `ProcessChecksInCoreAgentEnabled` option in `global` section

### PR DESCRIPTION
**Pull request by daniel-taf**
### What does this PR do?

This PR removes the `runProcessChecksOnCoreAgent` commandline flag and replaces any uses of it with the `processChecksInCoreAgentEnabled` option set in the `global` spec.

### Motivation

PROCS-4000

### Additional Notes

Anything else we should know when reviewing?

### Minimum Agent Versions

Are there minimum versions of the Datadog Agent and/or Cluster Agent required?

* Agent: vX.Y.Z
* Cluster Agent: vX.Y.Z

### Describe your test plan

Deploy the operator with the following config
```yaml
apiVersion: datadoghq.com/v2alpha1
kind: DatadogAgent
metadata:
  name: datadog
spec:
  override:
    nodeAgent:
      image:
        name: agent
        tag: 7.57.0
  global:
    processChecksInCoreAgentEnabled: true
# ...
  features:
    liveProcessCollection:
      enabled: true
```
Verify the process-agent container is not brought up
```
❯ kubectl get pod <pod_name> -o json | jq '.spec.containers[].name'
"agent"
"trace-agent"
```

### Checklist

- [x] PR has at least one valid label: `bug`, `enhancement`, `refactoring`, `documentation`, `tooling`, and/or `dependencies`
- [x] PR has a milestone or the `qa/skip-qa` label

### CODE REVIEW

**Review by estherk15 — APPROVED** [review]

File: api/datadoghq/v2alpha1/datadogagent_types.go [hunk 1]
```diff
@@ -1285,6 +1285,12 @@ type GlobalConfig struct {
 	// Configure the secret backend feature https://docs.datadoghq.com/agent/guide/secrets-management
 	// See also: https://github.com/DataDog/datadog-operator/blob/main/docs/secret_management.md
 	SecretBackend *SecretBackendConfig `json:"secretBackend,omitempty"`
+
+	// Configure whether the process Agent or core Agent collects process and/or container information (Linux only).
+	// (Requires Agent 7.57.0+)
+	// Default: 'false'
+	// +optional
+	ProcessChecksInCoreAgentEnabled *bool `json:"processChecksInCoreAgentEnabled,omitempty"`
```
Comment (celenechang) [inline api/datadoghq/v2alpha1/datadogagent_types.go:1293]: Two suggestions for alternate parameter names:
- `DisableProcessAgent`
- `RunProcessChecksInCoreAgent`

File: api/datadoghq/v2alpha1/datadogagent_types.go [hunk 2]
```diff
@@ -1285,6 +1285,12 @@ type GlobalConfig struct {
 	// Configure the secret backend feature https://docs.datadoghq.com/agent/guide/secrets-management
 	// See also: https://github.com/DataDog/datadog-operator/blob/main/docs/secret_management.md
 	SecretBackend *SecretBackendConfig `json:"secretBackend,omitempty"`
+
+	// Configure whether the process Agent or core Agent collects process and/or container information (Linux only).
```
Comment (celenechang) [inline api/datadoghq/v2alpha1/datadogagent_types.go:1289]: Could we also include that the Process Agent won't run if true?

Comment (daniel-taf) [reply api/datadoghq/v2alpha1/datadogagent_types.go:1272]: Other checks can run on the process agent and will still run in those cases. Updated the description to reflect that.

File: api/datadoghq/v2alpha1/datadogagent_types.go (hunk 1 above)
Comment (daniel-taf) [reply api/datadoghq/v2alpha1/datadogagent_types.go]: Went with `RunProcessChecksInCoreAgent`

**Review by levan-m — APPROVED** [review]

File: api/datadoghq/v2alpha1/datadogagent_types.go (hunk 2 above)
Comment (celenechang) [reply api/datadoghq/v2alpha1/datadogagent_types.go:1289]: Thanks! One more nit, I wonder if we should capitalize "core|process Agent"? It seems we do elsewhere in our docs cc @estherk15

Comment (estherk15) [reply api/datadoghq/v2alpha1/datadogagent_types.go:1289]: @celenechang I just did a search and we capitalize Process Agent, but don't capitalize core Agent. 🤷‍♀️

Comment (daniel-taf) [reply api/datadoghq/v2alpha1/datadogagent_types.go:1289]: I updated it to capitalize the "Process Agent" and left "core Agent" as it is.

Comment (celenechang) [reply api/datadoghq/v2alpha1/datadogagent_types.go]: Thanks @daniel-taf . I'll approve, but no promises I won't change "core" later..

**Review by celenechang — APPROVED** [review]

**Review by levan-m — APPROVED** [review]

**Pull request merged**